switchtec.c

```diff
@@ -1022,7 +1032,11 @@ static void enable_link_state_events(struct switchtec_dev *stdev)
 static void stdev_release(struct device *dev)
 {
 	struct switchtec_dev *stdev = to_stdev(dev);
-
+	if (stdev->dma_mrpc){
+		dma_free_coherent(&stdev->pdev->dev, sizeof(*stdev->dma_mrpc),
+				stdev->dma_mrpc, stdev->dma_mrpc_dma_addr);
+		iowrite32(0, &stdev->mmio_mrpc->dma_en);
```

Comment:
I think it's best we also clear the address as well as the enable.

Reply:
confirm that device side treat dma_mrpc_dma_addr 0 as a invalid address value, none dma operation will executed in 0 value case. So we can deinit the register by 0.
But from PCIe aspect, it is not explicit said in spec, the address 0 is a invalid field  in memory write tlp. This is why I hesitate to add clear the address by 0, but just disable dma function

Reply:
If something goes wrong I'd rather see and have to debug a write to an invalid address of zero than to a valid physical address that the OS isn't expected. The former will at worst produce an error while the latter will be scribbling on random memory in the system which would be a nightmare if it has to be debugged.